pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -450,3 +487,74 @@ func (r *SubnetPortReconciler) getLabelsFromVirtualMachine(ctx context.Context,
 	log.Info("got labels from virtualmachine for subnetport", "subnetPort.UID", subnetPort.UID, "vmName", vmName, "labels", vm.ObjectMeta.Labels)
 	return &vm.ObjectMeta.Labels, nil
 }
+
+func (r *SubnetPortReconciler) addressBindingMapFunc(ctx context.Context, obj client.Object) []reconcile.Request {
+	ab, ok := obj.(*v1alpha1.AddressBinding)
+	if !ok {
+		log.Info("Invalid object", "type", reflect.TypeOf(obj))
+		return nil
+	}
+	spList := &v1alpha1.SubnetPortList{}
+	spIndexValue := fmt.Sprintf("%s/%s", ab.Namespace, ab.Spec.VMName)
+	err := r.Client.List(context.TODO(), spList, client.MatchingFields{util.SubnetPortNamespaceVMIndexKey: spIndexValue})
+	if err != nil || len(spList.Items) == 0 {
+		log.Error(err, "Failed to list SubnetPort from cache", "indexValue", spIndexValue)
+		return nil
+	}
+	if ab.Spec.InterfaceName == "" {
+		if len(spList.Items) == 1 {
```

Comment:
Is it possible to check whether subnet port already got external address before we trigger a new event?

Reply:
Make sense. Added this check.